Code/Framework/AzFramework/Platform/Common/Xcb/generate_splash_screen_header.cmake

```diff
@@ -0,0 +1,34 @@
+# If not set explicitly, look for a splash image in the project directory
+if(NOT DEFINED SPLASH_FILE)
+    # start by looking for file in the project directory
+    if(EXISTS ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+        set(SPLASH_FILE ${PROJECT_SOURCE_DIR}/Resources/Splash.png)
+    endif()
+    # if not found, assume it's engine centric build
+    if(DEFINED LY_PROJECTS)
+        if(EXISTS ${LY_PROJECTS}/Resources/Splash.png)
+            set(SPLASH_FILE ${LY_PROJECTS}/Resources/Splash.png)
+        endif()
+    endif ()
+endif()
+# If value is set, check if it exists and generate the header file
+if(DEFINED SPLASH_FILE)
+    file(TO_CMAKE_PATH ${SPLASH_FILE} SPLASH_FILE_RESOLVED_PATH)
+    # check if it's a png file
+    if(NOT ${SPLASH_FILE_RESOLVED_PATH} MATCHES ".*\\.png")
+        message(WARNING "Splash file must be a PNG image, omitting splash screen generation.")
+    else()
+        if(EXISTS ${SPLASH_FILE_RESOLVED_PATH})
+            execute_process(COMMAND ${CMAKE_COMMAND} -E copy ${SPLASH_FILE_RESOLVED_PATH} ${CMAKE_BINARY_DIR}/Splash.png)
+            file(REMOVE ${CMAKE_BINARY_DIR}/Splash.h)
+            add_definitions(-DSPLASH_IMAGE_EXISTS)
+            execute_process(COMMAND xxd -i Splash.png Splash.h
```

Comment:
If `xxd` is not installed, this process will fail. You should handle this situation and present an appropriate warning (and undefined `SPLASH_FILE` in this case as well)